cmd/okctl/hooks/state.go

```diff
@@ -230,6 +230,29 @@ func PurgeRemoteState(o *okctl.Okctl) RunEer {
 	}
 }
 
+// VerifyClusterExistsInState ensures we have a cluster in the state before continuing
+// This could happen if maintenance mode was not run after upgrading
+// to 0.0.80 and we don't have a state.db in S3 - see #491
+func VerifyClusterExistsInState(o *okctl.Okctl) RunEer {
+	return func(_ *cobra.Command, _ []string) error {
+		handlers := o.StateHandlers(o.StateNodes())
+
+		_, err := handlers.Cluster.GetCluster(o.Declaration.Metadata.Name)
+		if err != nil {
+			log := logging.GetLogger(logComponent, "VerifyClusterExistsInState")
```

Comment:
Not really sure what best practice is here, but I usually initialize all that can be initialized before the return of the RunEer func. I.e between line 236 and 237 here. That's where I'd put everything that doesn't need execution JIT. I'd try moving the state handlers initialization there, and the logger initialization